designer/client/src/reducers/graph/reducer.ts

```diff
@@ -297,17 +298,29 @@ const graphReducer: Reducer<GraphState> = (state = emptyGraphState, action) => {
     }
 };
 
-const reducer = mergeReducers(graphReducer, {
+const reducer: Reducer<GraphState> = mergeReducers(graphReducer, {
     processToDisplay: {
         nodes,
     },
 });
 
-const undoableReducer = undoable(reducer, {
+const pick = <T extends NonNullable<unknown>>(object: T, props: NestedKeyOf<T>[]) => _pick(object, props);
+const omit = <T extends NonNullable<unknown>>(object: T, props: NestedKeyOf<T>[]) => _omit(object, props);
+
+const pickKeys: NestedKeyOf<GraphState>[] = ["fetchedProcessDetails", "processToDisplay", "unsavedNewName", "layout", "selectionState"];
+const omitKeys: NestedKeyOf<GraphState>[] = [
+    "processToDisplay.validationResult",
```

Comment:
I understand this concept and the problem we want to resolve here, but now, the validation error won't be clean after undo

To reproduce:
Remove the connection between nodes
Click undo button
The scenario is valid, but I can still see an error

Reply:
I was able to reproduce the situation, that after redo, validation is missing

https://github.com/TouK/nussknacker/assets/9945753/2df705c2-53eb-4a8d-9af1-b3bd51139034

Reply:
should work now

Reply:
Another visual thing, before process validation, nodes that had validation errors, were marked as invalid. It's not a big blocker, so if it's easy to fix, I would say, let's do it.


https://github.com/TouK/nussknacker/assets/9945753/02ad7cee-c93a-4a42-9e89-46bfbceae2aa

Reply:
better now?

Reply:
LGTM, thank you!